src/geocat/f2py/checks.py

```diff
@@ -0,0 +1,164 @@
+        Valid inputs are True or False
+
+    is_none : :class:`NoneType`:
+        Checking that user_variable is of class NoneType.
+        Valid input is None
+
+    comparison : :class:`int`:
+        Checks that user_variable is the correct type
+        ex: check(user_variable, comparison=
+
+    unchunked_dims : :class:`tuple`:
+        Checks for unchunked dimensions in user_variable
+        ex: check(user_variable, unchunked_dims=[3]) *must include brackets*
+
+    '''
+    if data_type is not None:
+        if not isinstance(user_variable, data_type):
+            raise TypeError(var_name + " failed data_type check.")
+
+    if dimensions is not None:
+        if not (user_variable.ndim == dimensions):
+            raise Exception(var_name + " failed dimensions check")
+
+    if shape is not None:
+        if not (user_variable.shape == shape):
+            raise Exception(var_name + " failed shape check")
+
+    if is_none is not None:
+        if is_none is True:
+            if user_variable is not None:
+                raise Exception(var_name + " is not None")
+        if is_none is False:
+            if user_variable is None:
+                raise Exception(var_name + " is None")
+
+    if min_dimensions is not None:
+        if not (user_variable.ndim >= min_dimensions):
+            raise Exception(var_name + " failed min_dimensions check")
+
+    if max_dimensions is not None:
+        if not (user_variable.ndim <= max_dimensions):
+            raise Exception(var_name + " failed max_dimensions check")
+
+    if comparison is not None:
+        if not (user_variable == comparison):
+            raise Exception(var_name + " failed comparison check")
+
+    #
+    # Xarray specific checks
+    #
+
+    if is_xarray is not None:
+        if is_xarray is True:
+            if not isinstance(user_variable, xr.DataArray):
+                # print(var_name + " is not xarray.DataArray, reformatting...")
+                return False
+            else:
+                pass
+        if is_xarray is False:
+            if isinstance(user_variable, xr.DataArray):
+                pass
+                # raise Exception(var_name + " is an xarray.DataArray")
+            else:
+                # print(var_name + " is not xarray.DataArray, reformatting...")
+                return True
+
+    if unchunked_dims is not None:
+        for dim in unchunked_dims:
+            if len(user_variable.chunks[dim]) > 1:
+                raise Exception(var_name +
+                                " must not be chunked along dimension " +
+                                str(dim))
+
+    return True
+
+n = 127
+
+xi = np.linspace(0, n, num=n // 2 + 1, dtype=np.float64)
+yi = np.linspace(0, n, num=n // 2 + 1, dtype=np.float64)
+yi_reverse = yi[::-1].copy()
+xo = np.linspace(xi.min(), xi.max(), num=xi.shape[0] * 2 - 1)
+yo = np.linspace(yi.min(), yi.max(), num=yi.shape[0] * 2 - 1)
+
+xt = xi
+
+def dtype_2d(xi=None, yi=None):
```

Comment:
These two functions `dtype_2d ` and `dtype_1d ` are very problem-specific and do not look like commony-applicable. They may be given up for the time until we see multiple common cases that would need such checking functions.